### ISSUE TITLE
Idempotent Issues

### ISSUE THREAD

**Issue by cdenneen** [jenkinsci/puppet-jenkins#560]
- [ ] jenkins::user - #454 
```puppet
Notice: /Stage[main]/Profile::Jenkins/Jenkins::User[johndoe]/Jenkins::Cli::Exec[create-jenkins-user-johndoe]/Exec[create-jenkins-user-johndoe]/returns: executed successfully
```
- [ ] jenkins_num_executors
```puppet
Notice: /Stage[main]/Profile::Jenkins/Jenkins_num_executors[2]/ensure: created
```
- [ ] jenkins_security_realm
```puppet
Notice: /Stage[main]/Profile::Jenkins/Jenkins_security_realm[hudson.plugins.active_directory.ActiveDirectorySecurityRealm]/arguments: arguments changed ['domain.local', 'SITE', 'CN=svc.puppet,OU=Users,DC=domain,DC=local', {'encryptedValue' => 'ASASDJHAKSJDHASDASDAD=', 'legacyKey' => {'algorithm' => 'AES', 'destroyed' => 'false', 'encoded' => ['98', '-97', '9', '-1', '9', '-3', '3', '-52', '-60', '93', '31', '102', '41', '9', '-13', '26'], 'format' => 'RAW'}, 'plainText' => 'supersecret'}, 'AUTO'] to 'domain.local SITE CN=svc.puppet,OU=Users,DC=domain,DC=local supersecret dc01.domain.local,dc02.domain.local'
```
  Oddly the xml already has the domain controllers in it... even though this shows the before as it's missing

**Comment by cdenneen** [comment]
For Jenkins_Security_Realm need some way to compare (please excuse this crude mock up);

$get_security_realm returns hash
$get_security_realm[setSecurityRealm][0] == jenkins_security_realm['0']
AND
$get_security_realm[setSecurityRealm][0][0] = jenkins_security_realm['0'][arguments][0]
AND so on...
$get_security_realm[setSecurityRealm][0][plainText] = jenkins_security_realm['0'][arguments][4]

else 

set_security_realm

Basically if the individual parts match including the plainText password then the security realm doesn't require updating... This would probably work better than the current hash comparison since the legacyKey hash that returns in the middle of get_security_realm I don't believe is something that we can mock around as I'm not sure where those values are derived from.

**Comment by cdenneen** [comment]
jenkins_num_executors - not sure why this one isn't idempotent since get_num_executors returns the same value we are setting. Maybe some sort of comparison logic is missing, resulting in the constant change?
